frontend/sw360-portlet/src/main/webapp/html/utils/includes/attachmentsDetail.jsp

```diff
@@ -21,82 +23,155 @@
+			    <col style="width:  6%;" />
+			    <col style="width:  6%;" />
+			    <col style="width: 16%;" />
+			    <col style="width:  6%;" />
+			    <col style="width: 16%;" />
+			    <col style="width:  6%;" />
+			    <col style="width: 10%;" />
+		    </colgroup>
+			<thead>
+				<tr>
+					<th>
+						<sw360:DisplayDownloadAttachmentBundle attachments="${attachments}"
+		                                                       name="AttachmentBundle.zip"
+		                                                       contextType="${documentType}"
+		                                                       contextId="${documentID}"/>
+					</th>
+					<th>File name</th>
+					<th>Size</th>
+					<th>Type</th>
+					<th>Group</th>
+					<th>Uploaded by</th>
+					<th>Group</th>
+					<th>Checked by</th>
+					<th>Usage</th>
+					<th>Actions</th>
+				</tr>
+			</thead>
+			<tbody>
+			</tbody>
+		</table>
+
+		<script>
+			var attachmentJSON = [];
+
+			/* Print all attachment table data as array into the html page */
+	        <core_rt:forEach items="${attachments}" var="attachment">
+				attachmentJSON.push({
+					"fileName": "${attachment.filename}",
+					"size": "n/a",
+					"type": "<sw360:DisplayEnumShort value="${attachment.attachmentType}"/>",
+					"uploadedTeam": "<sw360:DisplayEllipsisString value="${attachment.createdTeam}"/>",
+					"uploadedBy": "<sw360:DisplayEllipsisString value="${attachment.createdBy}"/>",
+					"checkedTeam":  "<sw360:DisplayEllipsisString value="${attachment.checkedTeam}"/>",
+					"checkedBy":  "<sw360:DisplayEllipsisString value="${attachment.checkedBy}"/>",
+					"usage":  "n/a",
+					"actions": 	"<sw360:DisplayDownloadAttachmentFile attachment="${attachment}" contextType="${documentType}" contextId="${documentID}"/>",
+
+					"sha1": "${attachment.sha1}",
+					"uploadedOn": "${attachment.createdOn}",
+					"uploadedComment": "<core_rt:if test="${not empty attachment.createdComment}">Comment: <sw360:DisplayEllipsisString value="${attachment.createdComment}"/></core_rt:if>",
+					"checkedOn": "${attachment.checkedOn}",
+					"checkedComment": "<core_rt:if test="${not empty attachment.checkedComment}">Comment: <sw360:DisplayEllipsisString value="${attachment.checkedComment}"/></core_rt:if>",
+
+					"checkStatus": "${attachment.checkStatus}"
+				});
+			</core_rt:forEach>
+
+			/* Define function for child row creation, which will contain additional data for a clicked table row */
+			function createChildRow(rowData) {
+				var childHtmlString = '' +
+						'<div>' +
+							'<span class="dataTableChildRowCell" style="padding-right: 10px; width:  4%;"/>' +
+							'<span class="dataTableChildRowCell" style="padding-right: 50px; width: 36%;">' + rowData.sha1 + '</span>' +
+							'<span class="dataTableChildRowCell" style="padding-right: 30px; width: 22%;">' + rowData.uploadedOn + ' ' + rowData.uploadedComment + '</span>';
+				if (rowData.checkStatus === 'ACCEPTED') {
+					childHtmlString += '' +
```

Comment:
can you explain to me what's the purpose of concatenating with this empty string `'' +` all the time, please?